src/incremental/compareCFG.ml

```diff
@@ -65,11 +62,10 @@ let compareCfgs (module CfgOld : CfgForward) (module CfgNew : CfgForward) fun1 f
             (* TODO: don't allow pseudo return node to be equal to normal return node, could make function unchanged, but have different sallstmts *)
             if eq_node (toNode1, fun1) (toNode2, fun2) && eq_edge_list edgeList1 edgeList2 then
               begin
-                let notInSame = not (NTH.mem same (toNode1, toNode2)) in
-                let matchedAlready = NTH.fold (fun (toNode1', toNode2') _ acc ->
-                    acc || (Node.equal toNode1 toNode1' && not (Node.equal toNode2 toNode2'))) same false in
+                let notInSame = not (NH.mem !same.node1to2 toNode1 && Node.equal (NH.find !same.node1to2 toNode1) toNode2) in
+                let matchedAlready = NH.mem !same.node1to2 toNode1 && not (Node.equal (NH.find !same.node1to2 toNode1) toNode2) in
```

Comment:
These look awfully similar now and do the same hashtable lookup 4 times. It might be easier to just do a single `match NH.find_opt ... with ...`. Especially since it looks like all four combinations of the two booleans wouldn't even be possible.